### PR TITLE
Bump node versions and clarify clean shutdown in `README.md`

### PR DESCRIPTION
**Pull request by darinspivey**
**fix: `destroyOnReturn` backward compatibility for node**

Although `destroyOnReturn` was added only in versions >= 16,
the option cannot be set on earlier versions. This sets up
a conditional to send options to the `iterator` only if it
exists.

Semver: patch

---

**fix: Necessary changes for asyncIterators in recent version of node**

Since node:16, `asyncIterator` has added an option called
`destroyOnReturn` which will abort the stream if an iterator loop does
not cleanly fall through. In the case of `TailFile`, backpressure is
respected, and that iteration will return early causing an `ABORT_ERR`
to be thrown.  The option for `destroyOnReturn` is true by default, so
this change turns off that experimental feature.

Semver: minor

---

**fix: Fix flakey test**

This test behaves better by setting up the Promise early such that
there is no race condition between the event and appending to the
file.

Semver: patch

---

**fix: Bump node versions for CI**

Add newest versions of node to include 16 and 17.
For backward compatibility with older versions, leave
the test matrix for those version 10 & 12.

Fixes: #26
Semver: patch

---

**fix: README should explain a clean shutdown**

Since there are various events and ways to track when
`TailFile` is finished, be clear with an example of
what to do to shut down and track the last known file
position.

Fixes: #24

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,6 +147,46 @@ startTail().catch((err) => {
 })
 ```
 
+### Example for Clean Shutdown
+
+`TailFile` will call `flush()` when `quit()` is called.  Therefore, to exit  cleanly,
+one must simply await the `quit` call.  If the implementation wishes to keep track of
+the last position read from the file (for resuming in the same spot later, for example),
+then a simple listener can be added to always track the file postiion.  That way, when
```
Comment (mdeltito) [inline README.md:155]: `position`

File: Jenkinsfile [hunk 2]
```diff
@@ -134,7 +134,7 @@ pipeline {
 
       agent {
         docker {
-          image "us.gcr.io/logdna-k8s/node:12-ci"
+          image "us.gcr.io/logdna-k8s/node:17-ci"
```
Comment (evanlucas) [inline Jenkinsfile:137]: I would use 16-ci here since 17 is not LTS. It might still get breaking changes

File: Jenkinsfile [hunk 3]
```diff
@@ -102,7 +102,7 @@ pipeline {
...
```
Comment (evanlucas) [inline Jenkinsfile:105]: ditto, I'd use 16-ci here

**Review by mdeltito — APPROVED** [review]
LGTM, just 1 typo

File: README.md (hunk 1 above)
Comment (darinspivey) [reply README.md:155]: ack

File: Jenkinsfile (hunk 2 above)
Comment (darinspivey) [reply Jenkinsfile:137]: ack

File: Jenkinsfile (hunk 3 above)
Comment (darinspivey) [reply Jenkinsfile:105]: ack

File: Jenkinsfile (hunk 2 above)
Comment (darinspivey) [reply Jenkinsfile:137]: To that point, I'll set `.nvmrc` to 16 as well.  Thanks!

**Review by mdeltito — APPROVED** [review]

**Review by evanlucas — APPROVED** [review]

**Pull request merged**